src/components/endpoints/SetData.vue

```diff
@@ -0,0 +1,93 @@
+<script setup lang="ts">
+import { getState } from "@/stores";
+import Notifications from "@/components/shared/Notification.vue";
+import useNotifications from "@/compositions/useNotifications";
+import useWeb3 from "@/compositions/useWeb3";
+import UniversalProfile from "@lukso/universalprofile-smart-contracts/artifacts/UniversalProfile.json";
+
+const { notification, clearNotification, hasNotification, setNotification } =
+  useNotifications();
+const { contract } = useWeb3();
+
+const setData = async () => {
+  const from = getState("address");
+
+  if (!from) {
+    return setNotification("No from address", "danger");
+  }
+
+  const erc725yContract = contract(UniversalProfile.abi as any, from);
+  const key =
+    "0x5ef83ad9559033e6e941db7d7c495acdce616347d28e90c7ce47cbfcfcad3bc5"; // key hash of LSP3Profile
+  const value =
+    "0x6f357c6a70546a2accab18748420b63c63b5af4cf710848ae83afc0c51dd8ad17fb5e8b3697066733a2f2f516d65637247656a555156587057347a53393438704e76636e51724a314b69416f4d36626466725663575a736e35"; // encoded profile ipfs url
+  try {
+    await erc725yContract.methods.setData([key], [value]).send({
+      from,
+    });
+
+    setNotification(`Set data`, "info");
+  } catch (error) {
+    setNotification((error as unknown as Error).message, "danger");
+  }
+};
+
+const setPermissions = async () => {
+  const from = getState("address");
+
+  if (!from) {
+    return setNotification("No from address", "danger");
+  }
+
```

Comment:
It is not really a "from" here, it is the address of the smart contract you want to interact with. In the case of LYX transfer it makes sense to call it "from" but here it might get confusing.